exercises/03.assertions/05.solution.soft-assertions/README.mdx

```diff
@@ -78,13 +78,13 @@ This gives me an overview of the entire system, not just the first failed assert
 		this.state = 'cancelled'
 
 		const today = new Date()
-		today.setUTCDate(1)
-		today.setUTCMonth(
-			today.getUTCMonth() + 1 > 11 ? 0 : today.getUTCMonth() + 1,
-		)
-
-		if (today.getUTCMonth() === 0) {
-			today.setUTCFullYear(today.getUTCFullYear() + 1)
+		switch (this.kind) {
+			case "monthly":
+				today.setUTCMonth(today.getUTCMonth() + 1, 1)
+				break;
+			case "yearly":
+				today.setUTCFullYear(today.getUTCFullYear() + 1, 0, 1)
```

Comment:
I appreciate the realism here but the subscription logic is largely irrelevant for the exercise. For example, it might as well be that in this app the yearly subscription still expires on the first day of the next month once you cancel it, and you get refunded the unused credits. That doesn't really matter. 

But what I want to try to do with your feedback is make the actions of the exercise more streamlined. Thinking about that...